### PR TITLE
ENV/scm/git: rewrite in Bash

### PR DESCRIPTION
**Pull request by xu-cheng**

### CODE REVIEW

File: Library/ENV/scm/git [hunk 1]
```diff
@@ -1,71 +1,113 @@
...
-  exec path, *ARGV unless path == "/usr/bin/#{F}"
-end
-
-popup_stub = false
-if executable? "/usr/bin/xcode-select"
+  fi
+  exec "$@"
+}
+
+F="${0##*/}"
```
Comment (MikeMcQuaid) [inline Library/ENV/scm/git]: I know it's not your issue but a nicer variable name would be great here.

File: Library/ENV/scm/git [hunk 2]
```diff
@@ -1,71 +1,113 @@
...
+D="$(quiet_cd "${0%/*}/" && pwd -P)"
```
Comment (MikeMcQuaid) [inline Library/ENV/scm/git]: Again: I know it's not your issue but a nicer variable name would be great here.

**Comment by MikeMcQuaid** [comment]
Wow, nice turnaround on this 👍. Looks good to me. Does `shellcheck` seem OK? If so and if you've tested it: :shipit:

File: Library/ENV/scm/git (hunk 2 above)
Comment (xu-cheng) [inline Library/ENV/scm/git]: Would you mind to suggest a better name as naming is hard 😂

File: Library/ENV/scm/git (hunk 1 above)
Comment (MikeMcQuaid) [inline Library/ENV/scm/git]: `FILENAME`?

File: Library/ENV/scm/git (hunk 2 above)
Comment (MikeMcQuaid) [inline Library/ENV/scm/git]: `SELF_DIRECTORY`?

File: Library/ENV/scm/git (hunk 1 above)
Comment (xu-cheng) [inline Library/ENV/scm/git]: How about `SHIMS_FILE` and SHIMS_DIR` or `SCM_FILE` and `SCM_DIR`? `FILENAME` itself is too vague.

Comment (MikeMcQuaid) [inline Library/ENV/scm/git]: `SCM_FILE` 👍

**Pull request merged**